hawkbit-sdk/hawkbit-sdk-demo/src/main/java/org/eclipse/hawkbit/sdk/demo/device/DeviceApp.java

```diff
@@ -54,44 +52,38 @@ HawkbitClient hawkbitClient(
     }
 
     @Bean
-    DdiController device(
-            @Value("${hawkbit.device:controller-default}") final String controllerId,
-            @Value("${hawkbit.device.securityToken:}") final String securityToken,
-            final Tenant defaultTenant,
-            final Optional<UpdateHandler> updateHandler,
-            final HawkbitClient hawkbitClient) {
-        return new DdiController(
-                defaultTenant,
-                Controller.builder()
-                        .controllerId(controllerId)
-                        .securityToken(ObjectUtils.isEmpty(securityToken) ?
-                                (ObjectUtils.isEmpty(defaultTenant.getGatewayToken()) ? SetupHelper.randomToken() : securityToken) :
-                                securityToken)
-                        .build(),
-                updateHandler.orElse(null),
-                hawkbitClient).setOverridePollMillis(10_000);
+    DdiTenant ddiTenant(final Tenant defaultTenant,
+                        final HawkbitClient hawkbitClient) {
+        return new DdiTenant(defaultTenant, hawkbitClient);
     }
 
     @ShellComponent
     public static class Shell {
 
-        private final Tenant tenant;
+        private final DdiTenant ddiTenant;
+
         private final DdiController device;
-        private final HawkbitClient hawkbitClient;
 
         private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
 
-        Shell(final Tenant tenant, final DdiController device, final HawkbitClient hawkbitClient) {
-            this.tenant = tenant;
-            this.device = device;
-            this.hawkbitClient = hawkbitClient;
+        Shell(final DdiTenant ddiTenant, final Optional<UpdateHandler> updateHandler) {
+            this.ddiTenant = ddiTenant;
+            String controllerId = System.getProperty("demo.controller.id");
+            String securityToken = System.getProperty("demo.controller.securityToken");
```

Comment:
better get from spring properties - so it could be configured via application properties and so on